src/components/ui/productCard/ProductCard.tsx

```diff
@@ -1,19 +1,28 @@
 import React from "react";
+import { Link } from "react-router-dom";
 import "./productCard.scss";
 import { ProductCardProps } from "../../../types";
 
-export default function ProductCard(props: ProductCardProps) {
+export default function ProductCard({ dogy }: ProductCardProps) {
```

Comment:
you should expect not the dogy object but all of its properties

export default function ProductCard({id, image, ...} : ProductCardProps)

at the moment there is an unnecessary nesting level in the properties of this component